tests/cpp/tests/test_nodes_indexing.cpp

```diff
@@ -874,160 +892,126 @@ TEST_CASE("AdvancedIndexingNode") {
-                }
+                // as promised, but 120 values into the array initially
+                std::vector<double> values(2 * 3 * 5 * 4);
+                std::iota(values.begin(), values.end(), 0);
+                arr_ptr->initialize_state(state, values);
+
+                // dyn array starts empty
+                dyn_ptr->initialize_state(state, std::vector<double>{});
+
+                graph.initialize_state(state);
 
                 AND_WHEN("We mutate the main array, grow the indexing nodes and propagate") {
-                    std::vector<double> values(2 * 3 * 5 * 4);
-                    std::iota(values.begin(), values.end(), 0);
+                    // double the size of the array
                     arr_ptr->grow(state, values);
-                    // i_ptr -> {0, 1}
-                    // j_ptr -> {1, 2}
-                    dyn_ptr->grow(state, {0, 1, 3, 1, 2, 3});
+                    dyn_ptr->grow(state);
 
-                    std::vector<double> expected({4,  5,  6,  7,  24,  25,  26,  27,
-                                                  44, 45, 46, 47, 68,  69,  70,  71,
-                                                  88, 89, 90, 91, 108, 109, 110, 111});
+                    graph.propagate(state, graph.descendants(state, {arr_ptr, dyn_ptr}));
 
-                    arr_ptr->propagate(state);
-                    dyn_ptr->propagate(state);
-                    i_ptr->propagate(state);
-                    j_ptr->propagate(state);
-                    adv->propagate(state);
-                    val->propagate(state);
+                    AND_WHEN("We commit") {
+                        graph.commit(state, graph.descendants(state, {arr_ptr, dyn_ptr}));
 
-                    THEN("The state has the expected values and the diff is correct") {
-                        CHECK(adv->size(state) == 24);
-                        CHECK(std::ranges::equal(adv->view(state), expected));
-                        verify_array_diff({}, expected, adv->diff(state));
+                        THEN("The output is as expected") {
+                            CHECK(std::ranges::equal(
+                                    adv_ptr->view(state),
+                                    std::vector{0, 1, 2, 3, 20, 21, 22, 23, 40, 41, 42, 43}));
+                            // ArrayValidationNode checks most of the consistency etc
+                        }
                     }
 
-                    AND_WHEN("We mutate the main array again") {
-                        arr_ptr->commit(state);
-                        dyn_ptr->commit(state);
-                        i_ptr->commit(state);
-                        j_ptr->commit(state);
-                        adv->commit(state);
-                        val->commit(state);
-
-                        std::vector<double> new_expected = expected;
-
-                        arr_ptr->set(state, 47, -1);
```

Comment:
These kinds of updates (within the captured range and outside of it), after commit, are rather important to test as well. Would rather not see them deleted

Reply:
Added back in in 5b2c1ee